### PR TITLE
Ipynb parse

### PR DESCRIPTION
**Pull request by billbrod**
Addresses #38 , adding the ability to parse markdown cells for dois. It does this by repurposing the machinery for parsing docstrings for dois, creating "fake functions" to pass to it.

### CODE REVIEW

**Comment by billbrod** [comment]
I am not familiar with Windows, so I don't understand how to fix the test that's failing.

File: r2t2/docstring_parser.py [hunk 1]
```diff
@@ -44,10 +55,21 @@ def iter_extract_docstring_from_lines(
 def iter_extract_docstring_from_file(
     path: Union[str, Path]
 ) -> Iterable[CodeDocumentComment]:
-    return iter_extract_docstring_from_text(
-        Path(path).read_text(),
-        filename=str(path)
-    )
+    path = Path(path)
+    txt = path.read_text()
```
Comment (jezcope) [inline r2t2/docstring_parser.py]: The error on Windows seems to be a text decoding problem, and it's occurring here. Perhaps specifying the `encoding=` parameter to [Path.read_text()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) would solve the problem?

**Review by jezcope — COMMENTED** [review]
I've had a quick look at the test log and made a suggestion — let us know if it helps!

File: r2t2/docstring_parser.py (hunk 1 above)
Comment (billbrod) [reply r2t2/docstring_parser.py]: Looks like specifying it as `utf-8` fixed the issue, thanks!

File: r2t2/__main__.py [hunk 2]
```diff
@@ -90,6 +96,15 @@ def run(args: argparse.Namespace):
 
     if os.path.isdir(args.target) or args.static:
         locate_references(args.target, encoding=args.encoding)
+        if args.notebook:
+            if not args.target.endswith('.ipynb'):
+                raise Exception("If --notebook flag is passed, target must be a"
```
Comment (valentinsulzer) [inline r2t2/__main__.py:101]: is there a test for this exception?

**Review by valentinsulzer — APPROVED** [review]
Looks good, thanks!

File: r2t2/__main__.py (hunk 2 above)
Comment (billbrod) [reply r2t2/__main__.py:101]: There is not -- adding it now.

**Comment by dalonsoa** [comment]
@all-contributors please add @billbrod for code

**Pull request merged**